Add nextfloat(::BigFloat, n), prevfloat(::BigFloat,n) (#31310)
JuliaLang/julia: NEWS.md

    `findall`, `findfirst`, `argmin`/`argmax` and `findmin`/`findmax` to work with these
    objects, returning the index of matching non-missing elements in the parent ([#31008]).

>   * `inv(::Missing)` has now been added and returns `missing` ([#31451]).

Reviewer: ```suggestion
  * `inv(::Missing)` has now been added and returns `missing` ([#31408]).
```
Rebase artifact? This shouldn't be changing the issue number for a different NEWS entry.

Author: Actually, I worked on that issue. And while I was adding `news` entry for the current issue, I realized that I had made a mistake in `inv` issue. That's I made the change.

Reviewer: Ah, ok. That should in principle be done in a separate commit but probably not worth it.

>   * `nextfloat(::BigFloat, n::Int)` and `prevfloat(::BigFloat, n::Int)` have now been added

Reviewer 2: nitpick, if you happen to re-push before merge: `n::Integer`

Reviewer: ```suggestion
  * `nextfloat(::BigFloat, n::Integer)` and `prevfloat(::BigFloat, n:: Integer)` methods have been added
```

Reviewer 3: @narendrakpatel is in class and will push this modification  when he gets home.

Reviewer 3: or -- just go ahead and make the change -- we all agree

Author: I am back. I will make the changes right away.

Reviewer 3: do you know how to "squash"? I don't.

Author: For now, I will just use `git commit --amend` but you can squash commit in the following way:
* `git rebase -i HEAD~n`
* replace `pick` with `squash`

Reviewer: You don't have to do anything. It can be squashed by the person doing the merge.

Reviewer: (But yes, it can be done by you as well with the appropriate git incantations.)

* A no-argument construct to `Ptr{T}` has been added which constructs a null pointer ([#30919])
* `strip` now accepts a function argument in the same manner as `lstrip` and `rstrip` ([#31211])
* `mktempdir` now accepts a `prefix` keyword argument to customize the file name ([#31230], [#22922])

> * `nextfloat(::BigFloat)` and `prevfloat(::BigFloat)` now uses the precision of the argument ([#31310])

Reviewer 2: nitpick, if you happen to re-push before merge: "uses" -> "use"

Reviewer: ```suggestion
* `nextfloat(::BigFloat)` and `prevfloat(::BigFloat)` now return a value with the same
   precision as their argument, which means that (in particular)
   `nextfloat(prevfloat(x)) == x` whereas previously this could result in a completely
   different value with a different precision ([#31310])
```

* A no-argument construct to `Ptr{T}` has been added which constructs a null pointer ([#30919])
* `strip` now accepts a function argument in the same manner as `lstrip` and `rstrip` ([#31211])
* `mktempdir` now accepts a `prefix` keyword argument to customize the file name ([#31230], [#22922])
* `nextfloat(::BigFloat)` and `prevfloat(::BigFloat)` now returns a value with the same precision

>     as their argument, which means that (in particular) `nextfloat(prevfloat(x)) == x` whereas

Reviewer: ```suggestion
  as their argument, which means that (in particular) `nextfloat(prevfloat(x)) == x` whereas
```

>     previously this could result in a completely different value with a different precision ([#31310])

Reviewer: ```suggestion
  previously this could result in a completely different value with a different precision ([#31310])
```

Author: It is just a markdown thing. I had `build` the docs and checked that it looks fine. Do you still want me to push to these changes?

Reviewer: Probably not worth it, although it does match the way the rest of the file is formatted. You can also just merge the suggested changes through the GitHub UI.

Author: Changed the files.
